bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/ListImpl.java

```diff
@@ -244,7 +252,11 @@ private void populateTagListItems() {
         if (ArrayUtils.isNotEmpty(tags)) {
             Page rootPage = getRootPage(PN_TAGS_PARENT_PAGE);
             if (rootPage != null) {
-                TagManager tagManager = resourceResolver.adaptTo(TagManager.class);
+                ResourceResolver tagResourceResolver = getTagResourceResolver();
+                if (tagResourceResolver == null) {
+                    tagResourceResolver = resourceResolver;
+                }
+                TagManager tagManager = tagResourceResolver.adaptTo(TagManager.class);
```

Comment:
pls close opened resourceResolver when it is not needed to avoid stack trace about unclosed sessions

Reply:
next to the unclosed resource resolver, I would raise the question if a dedicated service user is required to read the tags. Do we have to assume that the user executing the request is not able to read the tags?

Reply:
@lewandowskidawid, good catch! It's fixed in 228aa5784a710abf8dec4470d53c291a6c801a7e by using the try-with-resources syntax to benefit from the AutoClosable feature introduced in [1].

@joerghoh, as we want to keep the support for AEM 6.3 and 6.4 for the core-components the introduction of using a service resource resolver seems to be the only option as there is an explicit deny for everyone on the /etc path. The user which is used for this service is provided OOTB in AEM. Please let me know if you see another option.

[1] https://issues.apache.org/jira/browse/SLING-4805

Reply:
I don't think that on 6.3 /etc/ is already locked down, this has been introduced with the repo reorg on 6.4.
And on 6.4 the tags are relocated to /content/, so there shouldn't be any need to have read access to /etc/, thus this service user is not necessary. And if you run 6.4 still with tags in /etc, then you have take care about it on your own.

I do think that we get away without service user. Plus on the other hand side, the creation of a new session on every request will slow down the rendering quite a bit.

Reply:
![image](https://user-images.githubusercontent.com/689884/68593101-f9bff800-0494-11ea-9987-480acd88202f.png)
This is a local test publisher environment. Running on AEM 6.3 SP3 CFP4. We have this issue only for AEM 6.3. Maybe we can add an OSGi config option to enable the service user only on demand and use the resource resolver of the request as default?